cpp/src/io/text/byte_range_info.cpp

```diff
@@ -31,7 +31,7 @@ std::vector<byte_range_info> create_byte_range_infos_consecutive(int64_t total_b
   auto range_size = util::div_rounding_up_safe(total_bytes, range_count);
```

Comment:
Minor suggestion: Though not a part of this PR change set but `range_size` could be made const.

Reply:
Nvm, just checked that the return type of `div_rounding_up_safe` is constexpr anyway so auto should be good!

Reply:
Wait. The function `div_rounding_up_safe` is constexpr. Return type is `I`, here `int64_t`.  AFAIK, constexpr does not apply to return type.
`range_size` should be `int64_t`. 

https://godbolt.org/z/9seajen4e